specs/NestedFrame.md

```diff
@@ -0,0 +1,268 @@
+                    std::wstring message = NavigationStartingArgsToJsonString(
+                        m_webviewEventSource.get(), args,
+                        L"CoreWebView2Frame::NavigationStarting");
+                    PostEventMessage(message);
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+
+        frame2->add_ContentLoading(
+            Callback<ICoreWebView2FrameContentLoadingEventHandler>(
+                [this](ICoreWebView2Frame* sender, ICoreWebView2ContentLoadingEventArgs* args)
+                    -> HRESULT {
+                    std::wstring message = ContentLoadingArgsToJsonString(
+                        m_webviewEventSource.get(), args, L"CoreWebView2Frame::ContentLoading");
+                    PostEventMessage(message);
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+        
+        frame2->add_DOMContentLoaded(
+            Callback<ICoreWebView2FrameDOMContentLoadedEventHandler>(
+                [this](ICoreWebView2Frame* sender, ICoreWebView2DOMContentLoadedEventArgs* args)
+                    -> HRESULT {
+                    std::wstring message = DOMContentLoadedArgsToJsonString(
+                        m_webviewEventSource.get(), args,
+                        L"CoreWebView2Frame::DOMContentLoaded");
+                    PostEventMessage(message);
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+
+        frame2->add_NavigationCompleted(
+            Callback<ICoreWebView2FrameNavigationCompletedEventHandler>(
+                [this](
+                    ICoreWebView2Frame* sender,
+                    ICoreWebView2NavigationCompletedEventArgs* args) -> HRESULT {
+                    std::wstring message = NavigationCompletedArgsToJsonString(
+                        m_webviewEventSource.get(), args,
+                        L"CoreWebView2Frame::NavigationCompleted");
+                    PostEventMessage(message);
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+    }
+}
+
+void ScenarioWebViewEventMonitor::PostFrameCreatedEventMessage(wil::com_ptr<ICoreWebView2Frame> webviewFrame) {
+    wil::unique_cotaskmem_string name;
+    CHECK_FAILURE(webviewFrame->get_Name(&name));
+    auto frame5 = webviewFrame.try_query<ICoreWebView2Frame5>();
+    if (frame5)
+    {
+        UINT32 frameId = 0;
+        CHECK_FAILURE(frame5->get_FrameId(&frameId));
+    }
+
+    std::wstring message =
+        L"{ \"kind\": \"event\", \"name\": \"FrameCreated\", \"args\": {";
+    
+    message += L"\"frame\": " + EncodeQuote(name.get());
+    message += L",\"webview frame Id\": " + std::to_wstring((int)frameId) + L"}";
+    message +=
+            WebViewPropertiesToJsonString(m_webview.get());
+    message += L"}";
+    PostEventMessage(message);
+}
+```
+### C# Sample
+```c#

```

Comment:
The C# and C++ samples should be demonstrating the same scenario with the same code (or as close as practical between our C# and C++/COM APIs). That doesn't seem to be the case here.